abis_mapping/types/metadata.py

```diff
@@ -24,7 +31,11 @@ class TemplateMetadata(pydantic.BaseModel):
     sampling_type: str
     template_url: str
     schema_url: str
-    template_lifecycle_status: Literal["Beta", "Current", "Deprecated"]
+    template_lifecycle_status: Literal[
+        TemplateMetadataLifecycleStatus.BETA,
```

Comment:
That should be `TemplateMetadataLifecycleStatus.BETA.value` to get the string. As it is right now it is a Enum instance.
```
>>> TemplateMetadataLifecycleStatus.BETA
<TemplateMetadataLifecycleStatus.BETA: 'beta'>
>>> TemplateMetadataLifecycleStatus.BETA.value
'beta'
```

Reply:
I think the whole type hint can be simplified to use the enum directly:
 `template_lifecycle_status: TemplateMetadataLifecycleStatus`

Reply:
Agreed but it will raise an JSON serialisation error in partA for the the end-point `GET /api/v1/template/schema`.
We had this problem when we added in the schema the pydantic type `AnyUrl` in the schema.
https://youtrack.gaiaresources.com.au/youtrack/issue/BDRSPS-863/Bug-fix-end-point-GET-template-schema-returns-a-500-for-all-templates-but-the-incidental-v1
And @joecrowleygaia had to do a custom serializer for `AnyUrl`
https://github.com/gaiaresources/abis-mapping/pull/216/files#diff-84d114268c48535a67574c8a9eece95df0dd247cb6aad9599160e43dfbbf48b1R36

Reply:
Tried the enum and I get an error.
```
pydantic_core._pydantic_core.ValidationError: 1 validation error for TemplateMetadata
template_lifecycle_status
  Input should be an instance of TemplateMetadataLifecycleStatus [type=is_instance_of, input_value='deprecated', input_type=str]
    For further information visit https://errors.pydantic.dev/2.8/v/is_instance_of
  ```

Reply:
I know why, and I remember @chungvl telling me that. 
If we use enum we need to remove the [strict mode](https://docs.pydantic.dev/latest/concepts/strict_mode/) here
```python
md_class = types.metadata.TemplateMetadata.model_validate(md_dict, strict=True)
```
I'm stupid

Reply:
It's been a long day 😂

Reply:
BTW JSON serialisation doesn't work with enum